### ISSUE TITLE
Is there a way to implement "tags" to determine if an item can be dropped into a container?

### ISSUE THREAD

**Issue by Zachiah** [isaacHagoel/svelte-dnd-action#341]
I'm building a ui builder, and I have a more complicated use case than just the `type` prop can handle, and I would like to have something like tags instead of just the "group".

Example:
First off everything should be draggable into the sidebar,
`NavLink`s, `Image`s, and `Text` Should be able to put into the `Nav` component,
Everything except `NavLink`s should be able to get put into the `Content` component

I imagine this could be implemented by having tags on items and on the containers, and only allowing drags if the container has at least one tag that the item has

**Comment by isaacHagoel** [comment]
You can change the group dynamically based on the app state. Won't that
suffice?

**Comment by Zachiah** [comment]
I would have to change it base on which item is currently being dragged, is that possible?

**Comment by isaacHagoel** [comment]
Should be

**Comment by Zachiah** [comment]
Ok Thanks!

**Comment by Krischkros** [comment]
ngl its kinda inconvenient to prevent the drag based on type.
would be easier to cancel the whole event based on a (typecheck) condition or to have multiple allowed types as array.
i have to implement a wrapper just for this functionality which makes my application (game inventory ui) more complicated.
checked some related REPL based on my issue like this: 
https://svelte.dev/repl/aca4db78b927481db958df190b5d7d6e?version=3.38.2
don't get me wrong, it works.. my usecase is just different. 
i made a slot component which handles the dnd functionality and i have different item arrays for different grids (grid of slots). 
first grid pockets, second grid backpack, and so on.
and i have different slot types, one which only allows items of type weapon and slots just for items which allows every type.
i want to prevent drops on wrong slot types.

**Comment by isaacHagoel** [comment]
@Krischkros  no sure i follow. 
what i offered above is to dynamically set the `type` of each zone right when the dragging starts based on which item is being dragged.
can you further explain how this doesn't meet your needs?

**Comment by Krischkros** [comment]
To be fair, I have to admit that the examples pushed me so far away from the actual solution.
And the fact that no example was shown here kept me believing it was complicated but it is so simple.
My solution approach looks very unconventional, because I iterate through all my grids and during the trigger dragStarted set the "type" props to undefined if I want to allow everything or to some other value if I want to prevent it.
I had also tried to find another solution approach here.
Basically, you can ignore me here :)

**Comment by isaacHagoel** [comment]
@Krischkros if you get errors then it is a bug that needs to fixed. would be good if you could reproduce it in a REPL
